python/plugins/processing/algs/help/qgis.yaml

```
@@ -466,6 +466,8 @@ qgis:refactorfields: >
 
   The original layer is not modified. A new layer is generated, which contains a modified attribute table, according to the provided fields mapping.
 
+  Rows in orange are not allowed to be NULL in the layer used as target. However, since the algorithm creates a new layer with no constraints, this can be taken only as a hint.
```

```suggestion
  Rows in orange have a NOT NULL constraint in the layer from which these fields were loaded.
  Treat this information as a hint during configuration. No constraints will be added on an output layer nor will this be checked or enforced by the algorithm.
```

I think that's slightly more on topic.

---

Added a very similar text in the end, but now not only limited to NOT NULL constraints.